### PR TITLE
settings menu

### PR DESCRIPTION
**Pull request by Alexgamer470**
this pr includes the structure of the settings menu, a little cleanup and renaming of some variables.

### CODE REVIEW

File: menu/menu.cpp
Comment (MCWertGaming) [inline menu/menu.cpp:94]: typo

File: menu/menu.cpp
Comment (MCWertGaming) [inline menu/menu.cpp:220]: Is this important?

File: menu/menu.hpp
Comment (MCWertGaming) [inline menu/menu.hpp:26]: Shouldn't we turn these into an enum or something like that?

File: menu/visual.cpp
Comment (MCWertGaming) [inline menu/visual.cpp:98]: Place holder text

**Review by staxfur — CHANGES_REQUESTED** [review]
I have noted a few things, overall it looks fine though ^^

File: menu/menu.cpp
Comment (Alexgamer470) [reply menu/menu.cpp:200]: Yes. As soon as we add the control settings to this sub-menu, we have to update the settings description so that the player knows what the settings change in the game (like we did in other sub menus as well). And the `anim::snakeControls()`  function is for drawing this sub-menu's content (aka. settings).

In summary: Yes, it is TODO. I just commented it out, so that I exactly know where to continue.

File: menu/menu.cpp
Comment (Alexgamer470) [reply menu/menu.cpp:220]: Yes, this will be the function for opening/interacting with the controls sub-menu.

File: menu/visual.cpp
Comment (Alexgamer470) [reply menu/visual.cpp:98]: Yes, I did that so that I have three different options to choose from. Should I remove it?

File: menu/menu.hpp
Comment (MCWertGaming) [reply menu/menu.hpp:26]: Oki I'll make a PR

File: menu/visual.cpp
Comment (MCWertGaming) [reply menu/visual.cpp:98]: I mean will we add another option there?

Comment (MCWertGaming) [reply menu/visual.cpp:98]: But I guess it's fine foe now ^^

Comment (Alexgamer470) [reply menu/visual.cpp:98]: Probably. Maybe sound or smth.

**Review by staxfur — DISMISSED** [review]
Looks nice

**Comment by Alexgamer470** [comment]
Before I merge a quick question @MCWertGaming, because think I made a mistake, but I'm not sure: I declared the function `void subMenu(int cursorState)` in `menu.hpp`, but as you can see in `menu.cpp` I wrote `void menu::menu::subMenu(int subSelectoion)`. So is it wrong/not ideal to do so and should I change it or not? If yes, what is the correct way of declaring a function in this case?

**Comment by MCWertGaming** [comment]
No, it's right the way you did it. The `menu::menu` is just the namespace and the class. In the .hpp files you are creating the prototypes, which basically just tell the compiler in other .cpp files that the function exists. While in the .cpp file you are actually declaring the function. On compile time the linker will then attach the function to where it's being called from.

The namespace and class are just for us, so we know where functions and methods are coming from / the class also manages our data.

**Comment by Alexgamer470** [comment]
Oh.. Because I just searched the web and compared with your function `void license_menu_toggle(bool&)` and I changed the declaration now to `subMenu(int)`. Should I revert the latest commit?

btw I just realized that `void license_menu_toggle(bool&)` should be renamed to `void toggleLicenseMenu(bool&)` what do you think?

**Comment by MCWertGaming** [comment]
On the Prototype you don't have to write the name of the parameter, I have started doing it though, because when you are including the header where the function is prototyped and then use the autocompletion by your IDE, it'll only show your prototype, without the name of the variable. In CPP-Terminal this mean that your saw `Term::Terminal(bool, bool, bool, bool)` while I had to always check what which bool exactly did.

For the function names, I usually use snake_case over cammelCase because the STL does that as well. In languages like C# and GO I'm using CamelCase usually. It doesn't matter much though.

the & doesn't make sense for values of the type `bool` or `int` since it's creating a pointer which takes up more space than the actual variale that otherwise would just be copied directly.

**Review by staxfur — APPROVED** [review]

**Comment by MCWertGaming** [comment]
You can merge this when it's done ^^

**Review by staxfur — APPROVED** [review]

**Pull request merged**
